### ISSUE TITLE
With 6.0.300-preview.22204.3, warning NU5104 occurs when packing a F# project

### ISSUE THREAD

**Issue by WenJunJi01** [dotnet/sdk#24864]
**Environment:** .NET SDK 6.0.300-preview.22204.3 (runtime-6.0.3)

**Repro steps:**
1. Install VS 17.3.0 Preview 1.0 [32413.229.main] with inserted SDK 6.0.300-preview.22204.3 
2. Create a new f# project in CLI
3. Run _dotnet pack_

**Expected result:**
The f# project should package successfully without any errors and warnings.

**Actual result:**
Warning "NU5104: A stable release of a package should not have a prerelease dependency." occurs.
![image](https://user-images.githubusercontent.com/87680983/163351698-d7786aff-4d17-4895-8383-68956bedfaba.png)

**Dotnet info:**
C:\Users\v-wenjj>dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.300-preview.22204.3
 Commit:    4f23310c97

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.300-preview.22204.3\

Host (useful for support):
  Version: 6.0.3
  Commit:  c24d9a9c91

.NET SDKs installed:
  6.0.300-preview.22204.3 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by WenJunJi01** [comment]
The same issue also repros with .NET 7 Preview 4 SDK installed.
![image](https://user-images.githubusercontent.com/87680983/163365799-c0482cc0-33a3-477b-b1ff-7a319fa43411.png)
Dotnet info:
![image](https://user-images.githubusercontent.com/87680983/163366067-7185f45e-9c44-4cb2-9c6e-a9cc654ad350.png)

**Comment by baronfel** [comment]
@KevinRansom this is to be expected with preview SDK releases, right? since we're including preview FSharp.Core packages _in_ the SDK ahead of publishing stable versions to NuGet.org?

**Comment by WenJunJi01** [comment]
Also repro with .NET 7 Preview 5 (SDK Version 7.0.100-preview.5.22258.1)
![image](https://user-images.githubusercontent.com/87680983/167761024-36162f17-d588-40ac-8dea-8687bfaee821.png)
![image](https://user-images.githubusercontent.com/87680983/167761156-8e81badd-65ad-4599-932b-beb39e287eaf.png)

**Comment by baronfel** [comment]
I don't think this is an actual error: FSharp.Core is bundled as a preview version with each preview release of the SDK, and this is just appearing because the default version applied to a package when built is `1.0.0`, which triggers the NuGet version validation.  I believe that:

* when building/testing preview SDKs for F# projects, the FSharp.Core version will always be preview, and therefore either a) NU5104 should be ignored for FSharp.Core or b) the version of the test project used for packaging should be set to a preview version
* when building/testing stable SDKs for F# projects, the FSharp.Core version should always be stable, and therefore we should not see this issue.

**Comment by baronfel** [comment]
Tagging @KevinRansom for confirmation.
